plasmapy/particles/ionization_state.py

```diff
@@ -387,11 +381,10 @@ def __getitem__(self, value) -> List[IonicLevel]:
                     number_density=self.number_densities[Z],
                     T_i=self.T_i[Z],
                 )
+            elif same_element and same_isotope:
+                raise ChargeError("No charge number provided.")
             else:
-                if not same_element or not same_isotope:
-                    raise ParticleError("Inconsistent element or isotope.")
-                elif not has_charge_info:
-                    raise ChargeError("No charge number provided.")
+                raise ParticleError("Inconsistent element or isotope.")
```

Comment:
```python
            elif not (same_element and same_isotope):
                raise ParticleError("Inconsistent element or isotope.")
            else:
                raise ChargeError("No charge number provided.")
```

I suggest this because the `elif` condition is more closely related to `ParticleError` than `ChargeError`.  As an analogy, in the current version I feel the conditional is on apples but the error being thrown is about oranges.  This new arrangement just feels more readable to me.